controllers/imagerepository_controller.go

```diff
@@ -109,6 +164,9 @@ func (r *ImageRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	}
 
 	if imageRepository.Status.State == imagerepositoryv1alpha1.ImageRepositoryStateFailed {
+		// remove component from metrics map
```

Comment:
Do we need failures metric?

Reply:
that isn't for failures metric, but just for cleaning that map, in case full reconcile didn't finish but failed

Reply:
added